src/Microsoft.SqlTools.ServiceLayer/Profiler/ProfilerSessionMonitor.cs

```diff
@@ -50,35 +72,111 @@ public void AddSessionListener(IProfilerSessionListener listener)
         /// <summary>
         /// Start monitoring the provided sessions
         /// </summary>
-        public bool StartMonitoringSession(ProfilerSession session)
+        public bool StartMonitoringSession(string viewerID, IXEventSession session)
         {
             lock (this.sessionsLock)
             {
-                // start the monitoring thread 
+                // start the monitoring thread
                 if (this.processorThread == null)
                 {
-                    this.processorThread = Task.Factory.StartNew(ProcessSessions);;
+                    this.processorThread = Task.Factory.StartNew(ProcessSessions);
+                }
+
+                // create new profiling session if needed
+                if (!this.monitoredSessions.ContainsKey(session.ID))
+                {
+                    var profilerSession = new ProfilerSession();
+                    profilerSession.XEventSession = session;
+
+                    this.monitoredSessions.Add(session.ID, profilerSession);
                 }
 
-                if (!this.monitoredSessions.ContainsKey(session.SessionId))
+                // create a new viewer, or configure existing viewer
+                Viewer viewer;
+                if(!this.allViewers.TryGetValue(viewerID, out viewer))
```

Comment:
please add a ' ' between the 'f' and '(' characters in the `if` statement per .Net framework coding convention guidelines.  Please apply this change to all `if` statements in the the PR.